pkgs/development/python-modules/curl-cffi/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, setuptools
+, curl-impersonate-chrome
+, cffi
+, certifi
+}:
+
+buildPythonPackage rec {
+  pname = "curl-cffi";
+  version = "0.6.3";
+
+  src = fetchFromGitHub {
+    owner = "yifeikong";
+    repo = "curl_cffi";
+    rev = "v${version}";
+    hash = "sha256-VeBh5wp/VEMDGR2YK06w34hBv9qHIyA+EiZHrhEhAGw=";
+  };
+
+  patches = [
+    ./use-system-libs.patch
+  ];
+  buildInputs = [
+    curl-impersonate-chrome
+  ];
+
+  format = "pyproject";
+  build-system = [
+    setuptools
+  ];
+
+  nativeBuildInputs = [
+    cffi
+  ];
```

Comment:
Does cffi in `nativeBuildInputs` even do anything here?

Reply:
Cffi is used in build script so I guess it's needed for cross-compilation, though I haven't tested whether it works in a cross build yet

Reply:
Ah yep! I missed that. It is indeed needed. I've got some follow-up changes I'm cooking up a follow-up PR to fix cross-compilation for the package and fix darwin build as well

Reply:
Follow-up PR, if you were curious: #310386